src/brod_producer_buffer.erl

```diff
@@ -98,18 +110,31 @@ add(#buf{pending = Pending} = Buf, CallRef, Key, Value) ->
   Req = #req{ call_ref = CallRef
             , data     = fun() -> {Key, Value} end
             , bytes    = data_size(Key) + data_size(Value)
+            , ctime    = now_milli()
             , failures = 0
             },
   maybe_buffer(Buf#buf{pending = queue:in(Req, Pending)}).
 
 %% @doc Maybe (if there is any produce requests buffered) send the produce
 %% request to kafka. In case a request has been tried for more than limited
 %% times, and 'exit' exception is raised.
+%% Return `{Action, NewBuffer}', where `Action' can be:
+%% none:
+%%   There is no message to be sent, or allowed to be sent,
+%%   (e.g. due to onwire limit), caller should wait for the
+%%   next event to trigger a new `maybe_send' call.
+%%   Such event can either be a new produce request or a produce response.
+%% {delay, Time}:
+%%   Caller should retry after the returned milli-seconds.
+%% retry:
+%%   Failed to send a batch, caller should schedule a delayed retry.
```

Comment:
Perhaps just rename `none` to `ok`? As long as producer is concerned, it should be enough to know that either we successfully sent something, or got nothing to send. In both cases it's "ok".